src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -97,25 +104,89 @@ async def deploy_and_assert_grafana_agent(
 
     # Note(rgildein, dnplas): The grafana agent charm will go to BlockedStatus if it is not
     # related to any consumer (e.g. prometheus-k8s, grafana-k8s).
-    await model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=5 * 60)
+    # Note(rgildein): We use an idle_period of 60 so we can be sure that the targets have already
+    # been scraped.
+    await model.wait_for_idle(
+        apps=[GRAFANA_AGENT_APP], status="blocked", timeout=5 * 60, idle_period=60
+    )
 
 
-async def _check_metrics_endpoint(app: Application, metrics_endpoint: str) -> None:
-    """Check metrics endpoint accessibility.
+def _check_url(url: str, port: int, path: str) -> bool:
+    """Return False if port and path are not defined in url, True otherwise.
 
-    Checking accessibility of metrics endpoint from grafana-agent-k8s. If metrics endpoint is
-    defined as `*:5000/metrics` it will be changed to `<app-name>.<namespace>.svc:5000/metrics`.
+    Check that the expected port and path are in the url after parsing it.
     """
-    if metrics_endpoint.startswith("*"):
-        url = f"http://{app.name}.{app.model.name}.svc{metrics_endpoint[1:]}"
-    else:
-        url = f"http://{metrics_endpoint}"
+    output = urlparse(url)
+    return output.port == port and output.path == path
 
-    cmd = f"curl -m 5 -sS {url}"
-    grafana_agent_app = app.model.applications[GRAFANA_AGENT_APP]
-    log.info("testing metrics endpoint with cmd: `%s`", cmd)
-    for unit in grafana_agent_app.units:
-        await _run_on_unit(unit, cmd)
+
+async def _get_targets_from_grafana_agent(app: Application) -> None:
+    """Get defined metrics targets for application from grafana-agent-k8s API.
```

Comment:
```suggestion
    """Return a dictionary with data if the charm under test is listed in the targets; otherwise an empty dictionary. 
    
        This method makes a request to the grafana-agent-k8s targets endpoint
        to retrieve the state and data of the application under test and returns this
        data as a dictionary.

        Args:
          app(Application): this is ABC
```

Reply:
I changed the first line to use only `dict` instead of  `dictionary`, since it was too long.